mitosheet/src/mito/components/endo/ColumnHeader.tsx

```diff
@@ -82,7 +82,15 @@ const ColumnHeader = (props: {
     // Get the pieces of the column header. If the column header is not a MultiIndex header, then
     // lowerLevelColumnHeaders will be an empty array
     const { lowerLevelColumnHeaders, finalColumnHeader } = getColumnHeaderParts(columnHeader);
-    const matchesSearch = checkMatchesSearch(finalColumnHeader + '', props.uiState.currOpenSearch.searchValue);
+
+    // Check if the column header is a match of the search
+    const matchesSearch = props.uiState.currOpenSearch.matches.find((value) => {
```

Comment:
I am a little worried about the performance here on large datasets. Can we test with 100+ columns with 1500 rows, searching for a common value?

Reply:
It works without any issues on the `RockArt_600Columns` dataset and on a dataset w/ 100k rows so I think it should be ok